Core/PSPLoaders.cpp

```diff
@@ -144,6 +144,40 @@ void InitMemoryForGameISO(FileLoader *fileLoader) {
 	}
 }
 
+bool ReInitMemoryForGameISO(FileLoader *fileLoader) {
+	if (!fileLoader->Exists()) {
+		return false;
+	}
+
+	IFileSystem *fileSystem = nullptr;
+	IFileSystem *blockSystem = nullptr;
+
+	if (fileLoader->IsDirectory()) {
+		fileSystem = new VirtualDiscFileSystem(&pspFileSystem, fileLoader->Path());
+		blockSystem = fileSystem;
+	}
+	else {
+		auto bd = constructBlockDevice(fileLoader);
+		if (!bd)
+			return false;
+
+		ISOFileSystem *iso = new ISOFileSystem(&pspFileSystem, bd);
+		fileSystem = iso;
+		blockSystem = new ISOBlockSystem(iso);
+	}
+
+	if (pspFileSystem.GetSystem("disc0:") != pspFileSystem.GetSystem("umd0:")) {
```

Comment:
Even if they were previously the same (only happens for VirtualDiscFileSystem and headless), we need to remount all of them each time when using the prefix.  So we should just remove this if (or maybe use it for the delOldSystem parameter only.)

-[Unknown]